### DISCUSSION TITLE
Reset in Controller

### DISCUSSION THREAD

**Discussion by SergeyWebDeveloper** [react-hook-form/react-hook-form#4549] [Ideas] [answered]
Hello @bluebill1049. I have been using your wonderful library for a long time, but I am missing one very cool feature. Your documentation says: `If your form will invoke reset with default values, you will need to call useForm with defaultValues instead of setting the defaultValue on individual fields.` Example https://codesandbox.io/s/react-hook-form-controller-template-forked-gp4ot?file=/src/index.js
Playback steps:
step 1 - fill in the fields with any values
step 2 - click reset button (fields on the form have been reset)
step 3 - click Send button. In console we see old values.

Why can't we completely zero out the shape? It is not very convenient to specify default values in useForm({defaultValues: /.../}) every time if the form is very large. It seems to me that this feature is not enough for many, since many use material-ui, antd and other ui-libs. As an idea the following could be done:
```
reset({},{
  useDefaultValuesInFields: true
})
```
and the defaultValues from the fields would be pulled in state form.

**Comment by bluebill1049** [comment, marked as answer]
Thanks for the feedback, however, `defaultValues` is actually the preferred place for your defaultValue, it the source of truth for hook form check dirty, reset and etc. I would recommend moving into `defaultValues` instead.

**Reply by SergeyWebDeveloper** [reply]
thank you for your feedback. tell when to expect a stable version react-hook-form v.7?

**Reply by bluebill1049** [reply]
it's already in RC, will be released this week
